cmd/poller/poller.go

```diff
@@ -1363,6 +1370,55 @@ func (p *Poller) logPollerMetadata() (map[string]*matrix.Matrix, error) {
 	return nil, nil
 }
 
+func (p *Poller) sendHarvestVersion() error {
+	var (
+		poller     *conf.Poller
+		connection *rest.Client
+		err        error
+	)
+
+	// connect to the cluster and retrieve the system version
+	if poller, err = conf.PollerNamed(opts.Poller); err != nil {
+		return err
+	}
+	timeout, _ := time.ParseDuration(rest.DefaultTimeout)
+	if connection, err = rest.New(poller, timeout, p.auth); err != nil {
+		return err
+	}
+	err = connection.Init(2)
+	if err != nil {
+		return err
+	}
+
+	// Check if the cluster is running ONTAP 9.11.1 or later
+	// If it is, send a harvestTag to the cluster to indicate that Harvest is running
+	// Otherwise, do nothing
+
+	ontapVersion, err := goversion.NewVersion(connection.Cluster().GetVersion())
+	if err != nil {
+		return err
+	}
+
+	if ontapVersion.LessThan(goversion.Must(goversion.NewVersion("9.11.1"))) {
+		return err
+	}
+
+	// Send the harvestTag to the ONTAP cluster including the OS name, sha1(hostname), Harvest version, and max RSS in MB
+	osName := collector.GetOSName()
```

Comment:
In this case, it won't be an issue since this value is defined by Go. You can see the full list via `go tool dist list`. The OS name is the part before the `/` since `go tool dist list` returns GOOS/GOARCH.

On Friday, I also verified the values from auto-support Platform.OS.